### ISSUE TITLE
Allow library views to be specified as combinations of other library views

### ISSUE THREAD

**Issue by adhawkins** [adhawkins/SimpleLibraryViews#3]
It would be useful to be able to specify a library view whose contents were based on other library views.

For example, if I had a library view called 'andy' with my favourite albums, and another called 'audiobooks' that contained my audiobooks, a library view to be used in combination with randmon mixes could be:

andy and not audiobooks

**Comment by JFritzle** [comment]
I really appreciate and use SimpleLibraryViews. Thank you.

So far, one single library is selected like this:
![SimpleLibraryView1](https://user-images.githubusercontent.com/62614244/228785603-5ff0d2cc-82a3-4267-9db1-666e23d3e232.png)
Perhaps a combination of several libraries could be selected like that?
![SimpleLibraryView2](https://user-images.githubusercontent.com/62614244/228785686-31153a46-6870-4b24-a3c8-88d388bd57cd.png)

**Comment by adhawkins** [comment]
> I really appreciate and use SimpleLibraryViews. Thank you.
> 
> So far, one single library is selected like this: ![SimpleLibraryView1](https://user-images.githubusercontent.com/62614244/228785603-5ff0d2cc-82a3-4267-9db1-666e23d3e232.png) Perhaps a combination of several libraries could be selected like that? ![SimpleLibraryView2](https://user-images.githubusercontent.com/62614244/228785686-31153a46-6870-4b24-a3c8-88d388bd57cd.png)

I don't think that would work, as each player can only have a single Library View associated with it. That'd require a change to the server itself, rather than something this plugin could define.

The idea of this issue was to be able to define a library view called 'rock&pop&folk&country' that is a combination of the 'rock&pop' and 'folk&country' views. You would then be able to select this view as the one to apply to a player.

I'd have to have a think how that could be defined in the user interface, and how it could be coded. I'm afraid my Perl skills aren't really up to scratch!

**Comment by JFritzle** [comment]
Unfortunately, I have no Perl knowledge at all. So I don't know if it's even feasible. But let me explain my idea nevertheless.

Library Views are defined by files _[.]simple-library-views-\<library\>_ as before.
Library Views are created during the media scan as before.
Each player has associated one single Library View, as before. No changes to the server are required.

While currently one single Library View from list of available Library Views can be assigned to player,
a new _combined_ Library View is to be created _internally_ and _on the fly_ whenever library selection is changed.
This _combined_ Library View is to be assigned to player. 

_Combined_ Library View is to be created by simply merging selected and already existing Library Views.
After my opinion, this should not cost much time, as directories do not have to get scanned again,
but existing library entries have only to be merged together, sorted and duplicates removed.
While currently one Library View per player is kept track, an enumeration of selected Library Views per player must kept track.

By the way, it is already possible to define e.g. a library view called _rock&pop&folk&country_ that is a combination of the _rock&pop_ and _folk&country_ views with your existing plugin. Perhaps not as comfortable as in your idea.
Creating files _.simple-library-views-rock&pop_ and _.simple-library-views-rock&pop&folk&country_ in parent directory _rock&pop_ and creating files _.simple-library-views-folk&country_ and _.simple-library-views-rock&pop&folk&country_ in parent directory _folk&country_ and adding _rock&pop&folk&country_ to Library View names ...
![1](https://user-images.githubusercontent.com/62614244/228847500-4295b524-646a-4edb-b10c-bb2052077ea5.png)
... then after a media scan I am able to select combined Library View _rock&pop&folk&country_ containing sum of titles from _rock&pop_ and _folk&country_:
![2](https://user-images.githubusercontent.com/62614244/228848098-d7219a1a-29e7-4f0b-a6e4-d2d66c27eb7f.png)

But this is not very comfortable when building combinations of serveral libraries. Having n libraries, we have (2\**n)-1 possible combinations of 1, 2, 3, ... n libraries.
For n=4 libraries _rock&pop_, _folk&country_, _blues_ and _jazz_, we have already (2\**4)-1 = 16-1 = 15 possible library combinations.
Following your idea to be able to select a combination, it would be necessary to pre-define and create all Library View combinations of previously created single Library Views, a huge effort. Following my idea, desired combination of selected Library Views would be created on demand and on the fly and replaced when library selection is changed.

**Comment by adhawkins** [comment]
I see what you're saying. You would still end up with a large number of internal views created. For example, if you have 4 views, A, B, C, D. the views created would be:

A
B
C
D
A+B
A+C
A+D
B+C
B+D
C+D
A+B+C
A+B+D
A+C+D
B+C+D
A+B+C+D

This could soon get unwieldy in the interface where you apply the library view to a player. I'd have to have a think about the logic involved in creating them all also.

Not a bad idea, will have a think.
